pkg/operator/apis/monitoring/v1/http_types_test.go

```diff
@@ -0,0 +1,137 @@
+package v1
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/prometheus/prometheus/secrets"
+	"github.com/prometheus/prometheus/secrets/kubernetes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestClusterSecretKeySelector_toPrometheusSecretRef_PodMonitoring(t *testing.T) {
+	p := &PodMonitoring{
+		ObjectMeta: metav1.ObjectMeta{Namespace: "foo"},
+	}
+
+	t.Run("nil", func(t *testing.T) {
+		pool := PrometheusSecretConfigs{}
+		var c *ClusterSecretKeySelector
+
+		ref, err := c.toPrometheusSecretRef(p, pool)
+		if err != nil {
+			t.Fatalf("unexpected failure: %s", err)
+		}
+		if ref != "" {
+			t.Fatalf("expected empty ref, got %v", ref)
+		}
+		if len(pool.SecretConfigs()) != 0 {
+			t.Fatalf("expected no configs, got %v", pool.SecretConfigs())
+		}
+	})
+
+	// Empty or correct namespace.
+	for _, ns := range []string{"", p.Namespace} {
+		t.Run(fmt.Sprintf("namespace=%v", ns), func(t *testing.T) {
+			pool := PrometheusSecretConfigs{}
+			c := &ClusterSecretKeySelector{
+				Name:      "secret1",
+				Key:       "key1",
+				Namespace: ns,
+			}
+
+			ref, err := c.toPrometheusSecretRef(p, pool)
+			if err != nil {
+				t.Fatalf("unexpected failure: %s", err)
+			}
+			if exp, got := "foo/secret1/key1", ref; exp != got {
```

Comment:
nit: maybe we can extract this string format into a method? We currently have one but the format is slightly different (see `pathForSelector` in `operator_config.go`.

Reply:
I think we could do that, but I prefer not as for "expected" strings it's more readable and safer to not use dynamic code. Only 4 very simple references